stable_baselines3/common/save_util.py

```diff
@@ -174,32 +176,61 @@ def json_to_data(json_string: str,
     return return_data
 
 
-def save_to_zip_file(save_path: str, data: Dict[str, Any] = None,
-                     params: Dict[str, Any] = None, tensors: Dict[str, Any] = None) -> None:
+@functools.singledispatch
+def open_path(path, verbose=0, suffix=None) -> io.IOBase:
```

Comment:
These functions could use proper docstrings. This function specifically only "works" as intended if `path` is `io.IOBase`, no? I think that should be checked here.

Reply:
The function works with any object that supports the `io.IOBase` api regardless of whether it inherits from it so, due to duck-typing, I left it as generic as possible and checking only the `str` and `Path` cases.

Reply:
I see there is also "StringIO" which is subclass of "IOBase", but I do not think the stuff being stored here can be stored in text buffers. I was pondering if we should use BytesIO for that reason to make it clear it can not be _any_ IOBase, but something that handles bytes :)